app/src/androidTest/java/com/pileproject/drive/execution/NxtControllerTest.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (C) 2011-2015 PILE Project, Inc. <dev@pileproject.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pileproject.drive.execution;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import com.pileproject.drivecommand.machine.device.input.LineSensor;
+import com.pileproject.drivecommand.machine.device.input.SoundSensor;
+import com.pileproject.drivecommand.machine.device.input.TouchSensor;
+import com.pileproject.drivecommand.machine.device.output.Motor;
+import com.pileproject.drivecommand.model.nxt.NxtMachine;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.internal.util.reflection.Whitebox;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@RunWith(AndroidJUnit4.class)
+public class NxtControllerTest {
+
+    NxtMachine machine;
+    NxtController controller;
+
+    TouchSensor touchSensor;
+    SoundSensor soundSensor;
+    LineSensor lineSensor;
+
+    Motor leftMotor;
+    Motor rightMotor;
+
+    @Before
+    public void setUp() {
+        machine = mock(NxtMachine.class);
+        controller = new NxtController(machine);
+
+        touchSensor = mock(TouchSensor.class);
+        soundSensor = mock(SoundSensor.class);
+        lineSensor = mock(LineSensor.class);
+
+        leftMotor = mock(Motor.class);
+        rightMotor = mock(Motor.class);
+    }
+
+    private void setUpMotors() {
+        Whitebox.setInternalState(controller, "mLeftMotor", leftMotor);
+        doCallRealMethod().when(leftMotor).setSpeed(anyInt());
+        doCallRealMethod().when(leftMotor).getSpeed();
+
+        Whitebox.setInternalState(controller, "mRightMotor", rightMotor);
+        doCallRealMethod().when(rightMotor).setSpeed(anyInt());
+        doCallRealMethod().when(rightMotor).getSpeed();
+    }
+
+
+    // TODO: add tests related to preferences
```

Comment:
Yeah, I agree with the idea and I also considered it but I concluded doing it in this PR is too much. Let's make another PR for this.